pymc/gp/cov.py

```diff
@@ -47,26 +49,22 @@
 ]
 
 
-class Covariance:
-    r"""
-    Base class for all kernels/covariance functions.
+def _verify_scalar(value):
+    if (
+        isinstance(value, pytensor.compile.SharedVariable)
+        and value.get_value().squeeze().shape == ()
+    ):
+        return at.squeeze(value)
```

Comment:
This is slightly incorrect. If I pass `pt.shared(np.ones(1))` it will fail and not be captured by your error.

If I pass `pt.shared((1), shape=(1,))` then it will work.

I suggest just calling squeeze directly. Also inputs could be constants (e.g., from pm.ConstantData) and would not meet either branch, no?

Reply:
This functions is only used by exponentiated kernels now (which aren't usable by HSGP).  I tried to use `_verify_scalar` for other HSGP stuff earlier in this PR (that I cant recall) but it got refactored out. 

Thanks for catching this potential regression.  In practice though, I think its extremely unlikely anyone would do anything more sophisticated than
```
cov_func = eta**2 * pm.gp.cov.ExpQuad(1, ls)**2 # or cubed or something
```
`_verify_scalar` is only handling the power, `**2`.  I'm also OK with not checking/supporting type checking here.  Basically usage wise there's really no chance that exponent is anything other than "2" or maybe "3".  I've not see this really in the wild in models, more that its just kinda neat that you can exponentiate kernels (because products of kernels are kernels) so why not support it.   

How about I roll things back to how they are in master for this, which is, 
```python
    def __pow__(self, other):
        if (
            isinstance(other, pytensor.compile.SharedVariable)
            and other.get_value().squeeze().shape == ()
        ):
            other = at.squeeze(other)
            return Exponentiated(self, other)
        elif isinstance(other, Number):
            return Exponentiated(self, other)
        elif np.asarray(other).squeeze().shape == ():
            other = np.squeeze(other)
            return Exponentiated(self, other)

        raise ValueError("A covariance function can only be exponentiated by a scalar value")
```
The errors aren't captured, but users will see what happens when squeeze is attempted.

Reply:
No strong preference, but why is this not enough?

```python
def __pow__(self, other):
  other = as_tensor_variable(other).squeeze()
  if not other.ndim == 0:
    raise ValueError(...)
  return Exponentiated(self, other)
```

Reply:
alright that's why they pay you the big bucks, changed it to this